jq_array_parser.go

```diff
@@ -2,43 +2,46 @@ package opslevel_jq_parser
 
 import (
 	"encoding/json"
-	"strings"
+	"github.com/opslevel/opslevel-jq-parser/v2024/common"
 )
 
 type JQArrayParser struct {
-	programs []*JQFieldParser
+	programs []JQFieldParser
 }
 
-func NewJQArrayParser(expressions []string) *JQArrayParser {
-	programs := make([]*JQFieldParser, len(expressions))
+func NewJQArrayParser(expressions []string) JQArrayParser {
+	programs := make([]JQFieldParser, len(expressions))
 	for i, expression := range expressions {
 		programs[i] = NewJQFieldParser(expression)
 	}
-	return &JQArrayParser{
+	return JQArrayParser{
 		programs: programs,
 	}
 }
 
-func (p *JQArrayParser) Run(data string) []string {
-	output := make([]string, 0, len(p.programs))
```

Comment:
```
	output := make([]string, 0, len(p.programs))	
```

In the future, we should avoid adding the third argument here (capacity) - our code does not need to be this performant.

AFAICT there is actually a bug here where there are N programs, but more than N outputs because some of the responses can be arrays themselves. This would be more than the capacity we have.